### PR TITLE
Added convenience function for AJAX calls, put them in everywhere that fit

### PR DESCRIPTION
**Pull request by MattPutnam**
This has not been tested on live data yet, so don't merge until that can happen.

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs [hunk 1]
```diff
@@ -67,10 +67,10 @@
                     (aset xhr (name k) v))
                   xhr))
           call-on-done (fn []
-                         ((:on-done arg-map) {:xhr xhr
-                                              :status-code (.-status xhr)
-                                              :success? (and (>= (.-status xhr) 200)
-                                                             (< (.-status xhr) 300))}))]
+                         (on-done {:xhr xhr
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: What is the significance of changing `:on-done` to `on-done` ?

File: src/cljs/org/broadinstitute/firecloud_ui/page/method_repo.cljs [hunk 2]
```diff
@@ -59,15 +59,11 @@
...
-       {:on-done (fn [{:keys [success? xhr]}]
-                   (if success?
-                     (let [methods (utils/parse-json-string (.-responseText xhr))]
-                       (swap! state assoc :methods-loaded? true :methods methods))
-                     (swap! state assoc
-                            :error {:message (.-statusText xhr)})))
-        :canned-response {:responseText (utils/->json-string (create-mock-methods))
-                          :status 200
-                          :delay-ms (rand-int 2000)}}))})
+       {:on-success (fn [{:keys [parsed-response]}]
```
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/page/method_repo.cljs]: In clojure, does this mean we are overriding the `on-success` that exists in `call-ajax-orch` ?

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs (hunk 1 above)
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: This was just a little harmless oversight.  `(:on-done arg-map)` is already bound to `on-done` from the `let`, so no need to get it out of the map again.

File: src/cljs/org/broadinstitute/firecloud_ui/page/method_repo.cljs (hunk 2 above)
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/method_repo.cljs]: This pattern is supplying the `on-success` function to `call-ajax-orch`.

File: src/cljs/org/broadinstitute/firecloud_ui/utils.cljs (hunk 1 above)
Comment (rushtong) [inline src/cljs/org/broadinstitute/firecloud_ui/utils.cljs]: Yes - I see that now on line 55 of the new code.

**Comment by rushtong** [comment]
Tested all of the ajax calls - looks great!

**Comment by rushtong** [comment]
@MattPutnam - can you rebase? Feel free to merge after that.

**Pull request merged**